spec/lib/pdf_filler/md502_pdf_spec.rb

```diff
@@ -0,0 +1,145 @@
+          expect(pdf_fields["Text Box 96"]).to eq("5551234567")
+        end
+      end
+    end
+
+    context "mfj" do
+      let(:intake) { create(:state_file_md_intake, :with_spouse) }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          puts pdf_fields
+          expect(pdf_fields['Enter social security number']).to eq("400000030")
+          expect(pdf_fields["Enter spouse&apos;s social security number"]).to eq("600000030")
+          expect(pdf_fields["Enter your first name"]).to eq("Mary")
+          expect(pdf_fields["Enter your middle initial"]).to eq("A")
+          expect(pdf_fields["Enter your last name"]).to eq("Lando")
+          expect(pdf_fields["Enter Spouse&apos;s First Name"]).to eq("Marty")
+          expect(pdf_fields["Enter Spouse&apos;s middle initial"]).to eq("B")
+          expect(pdf_fields["Enter Spouse&apos;s last name"]).to eq("Lando")
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Yes"
+          expect(pdf_fields['Check Box - 3']).to eq "Off"
+          expect(pdf_fields["MARRIED FILING Enter spouse&apos;s social security number"]).to eq("")
+          expect(pdf_fields['Check Box - 4']).to eq "Off"
+          expect(pdf_fields['Check Box - 5']).to eq "Off"
+          expect(pdf_fields['6. Check here']).to eq "Off"
+        end
+      end
+    end
+
+    context "mfs" do
+      let(:intake) { create(:state_file_md_intake, :with_spouse, filing_status: "married_filing_separately") }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Enter social security number']).to eq("400000030")
+          expect(pdf_fields["Enter spouse&apos;s social security number"]).to eq("600000030")
+          expect(pdf_fields["Enter your first name"]).to eq("Mary")
+          expect(pdf_fields["Enter your middle initial"]).to eq("A")
+          expect(pdf_fields["Enter your last name"]).to eq("Lando")
+          expect(pdf_fields["Enter Spouse&apos;s First Name"]).to eq("Marty")
+          expect(pdf_fields["Enter Spouse&apos;s middle initial"]).to eq("B")
+          expect(pdf_fields["Enter Spouse&apos;s last name"]).to eq("Lando")
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Off"
+          expect(pdf_fields['Check Box - 3']).to eq "No"
+          expect(pdf_fields["MARRIED FILING Enter spouse&apos;s social security number"]).to eq("600000030")
+          expect(pdf_fields['Check Box - 4']).to eq "Off"
+          expect(pdf_fields['Check Box - 5']).to eq "Off"
+          expect(pdf_fields['6. Check here']).to eq "Off"
+        end
+      end
+    end
+
+    context "hoh" do
+      let(:intake) { create(:state_file_md_intake, :head_of_household) }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
```

Comment:
[DUST] maybe the description could mention that this test is for filing status since the pdf field names themselves are not descriptive?